source/grid/grid_in.cc

```diff
@@ -1635,41 +1638,50 @@ GridIn<dim, spacedim>::read_comsol_mphtxt(std::istream &in)
       else if (object_type == "edg")
         {
           AssertThrow(n_vertices_per_element == 2, ExcInternalError());
-          if (dim == 1)
-            cells.resize(n_elements);
-          else
-            subcelldata.boundary_lines.resize(n_elements);
         }
       else if (object_type == "tri")
         {
           AssertThrow(dim >= 2,
                       ExcMessage("Triangles should not appear in input files "
                                  "for 1d meshes."));
           AssertThrow(n_vertices_per_element == 3, ExcInternalError());
-          if (dim == 2)
-            cells.resize(n_elements);
-          else
-            subcelldata.boundary_quads.resize(n_elements);
+        }
+      else if (object_type == "quad")
+        {
+          AssertThrow(dim >= 2,
+                      ExcMessage(
+                        "Quadrilaterals should not appear in input files "
+                        "for 1d meshes."));
+          AssertThrow(n_vertices_per_element == 4, ExcInternalError());
         }
       else if (object_type == "tet")
         {
           AssertThrow(dim >= 3,
                       ExcMessage("Tetrahedra should not appear in input files "
                                  "for 1d or 2d meshes."));
           AssertThrow(n_vertices_per_element == 4, ExcInternalError());
-          if (dim == 3)
-            cells.resize(n_elements);
-          else
-            Assert(false, ExcInternalError());
+        }
+      else if (object_type == "prism")
+        {
+          AssertThrow(dim >= 3,
+                      ExcMessage("Prisms should not appear in input files "
+                                 "for 1d or 2d meshes."));
+          AssertThrow(n_vertices_per_element == 6, ExcInternalError());
```

Comment:
Could you add pyramids and hexahedra while you are here?

Reply:
The problem is that I have no example input file that contains these kinds of cells. I'm loath to add functionality that I can't test...

Reply:
then lets leave them out.